src/c2goto/library/time.c

```diff
@@ -6,10 +6,12 @@
 #include <time.h>
 #undef time
 
+time_t __VERIFIER_nondet_time_t();
+
 time_t time(time_t *tloc)
 {
-  time_t res;
-  if(!tloc)
+  time_t res = __VERIFIER_nondet_time_t();
+  if(tloc != NULL)
```

Comment:
No need to compare against NULL (btw, is NULL defined in this header)? `if(tloc)` should be enough.